src/scandeval/generation.py

```diff
@@ -165,79 +192,215 @@ def generate_single_iteration(
             The configuration of the dataset.
         benchmark_config:
             The configuration of the benchmark.
+        cache_path:
+            The path to the model output cache.
 
     Returns:
         A list of dictionaries containing the scores for each metric.
     """
-    # Tokens used in generation to know when generation is finished
-    stopping_criteria = get_generation_stopping_criteria(
-        tokenizer=tokenizer, model=model
-    )
+    # Load the model output cache
+    with cache_path.open() as f:
+        json_cache = json.load(f)
+        cache: dict[str, dict[str, CachedModelOutput]] = defaultdict(dict)
+        if not hasattr(sys, "_called_from_test"):
+            for key in json_cache:
+                for sub_key in json_cache[key]:
+                    cache[key][sub_key] = CachedModelOutput(**json_cache[key][sub_key])
+
+    # Initialise the cache for the current max tokens, if it doesn't exist already
+    max_tokens_str = str(dataset_config.max_generated_tokens)
+    if max_tokens_str not in cache:
+        cache[max_tokens_str] = dict()
```

Comment:
Not sure I get why it is created for the max token str?

Reply:
It was simply if the same prompt had two different completions, with different amounts of max tokens. For instance, if the same sample is used for two different tasks (say, sentiment and NER).

Reply:
After reconsidering, I ended up removing it after all, since I completely forgot about the prompt creation, which includes things like "\nSentiment:" at the end, for instance. This means that if the same sample is used for two different tasks, then their inputs would be different as well anyway.

Reply:
Makes sense